### PR TITLE
Preload schemas on startup for validation

### PR DESCRIPTION
**Pull request by brenetic**
## Preload schemas into JSON::Validator

Local referencing seems to be an issue with the json-schema gem. There are a number of issues open:

https://github.com/ruby-json-schema/json-schema/issues/190
https://github.com/ruby-json-schema/json-schema/issues/370
https://github.com/ruby-json-schema/json-schema/issues/411

To get around this we are preloading all the schemas into the `JSON::Validator` at startup using the `_name` property as the identifier for each schema. This should allow the `JSON::Validator` the ability to more easily find the necessary referenced schemas.

## Add unit test for MetadataSerialiser

This is a basic schema validation for the attributes that the MetadataSerialiser will be returning when requested.

Also add a fixture for a service. This is only for when creating a service and as such will not have the service_id or created_at as those are set by the metadata api before being returned.

## Use larger metadata fixture for integration tests

The integration tests now use the newer service and version fixtures. The version fixture has additional pages and a service id which is more like what the editor will be sending the metadata api.

## Bump public suffix to 4.0.6

### CODE REVIEW

File: spec/support/schema_matcher.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+RSpec::Matchers.define :match_schema do |schema|
+  match do |metadata|
+    ValidateSchema.validate(metadata, schema)
+  end
+end
```
Comment (tomas-stefano) [inline spec/support/schema_matcher.rb:5]: Cool, excellent idea. What's the failure message when test fail for this matcher?

**Review by tomas-stefano — APPROVED** [review]
LGTM 🚀

File: spec/support/schema_matcher.rb (hunk 1 above)
Comment (brenetic) [reply spec/support/schema_matcher.rb:5]: It bubbles up the message from the `json-schema` gem. Unfortunately the message can sometimes be a little hard to follow, but it at the very least references the object it was trying to check and the property it failed on

**Review by njseeto — APPROVED** [review]
⭐ 💯
